Placeholders (#343)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/all/labeling.mdx

<Vocabulary>

### Placeholders files

> Multiple JSON files where you can manage labels for your storefront, which includes localization. Each drop-in component has its own placeholders file (e.g., `cart.json`, `checkout.json`, `pdp.json`). See the [Placeholders files](/resources/placeholders/) for more information.

Reviewer: ```suggestion
Multiple JSON files to manage storefront labels, including localization. Each drop-in component has its own placeholders file (e.g., `cart.json`, `checkout.json`, `pdp.json`). See the [Placeholders files](/resources/placeholders/) for more information.
```

### Language objects

A JavaScript object (named `langDefinitions` by convention) that contains key-value pairs for UI text labels in a specific language. The object is used to override the default dictionary with the placeholder file's keys and values.

### Labeling

> Add or change UI text labels within Commerce drop-in components. Merchants can update storefront UI labels by editing the placeholders file in their content directory.

Reviewer: ```suggestion
To add or change UI text labels within Commerce drop-in components. Merchants use placeholer files to quickly change the default UI labels for drop-in components.
```

<Callouts>

> 1. **Placeholder files**. Merchants can quickly update the labels in their storefront from JSON files: `placeholders`, which are organized by drop-in component (e.g., `cart.json`, `checkout.json`, `pdp.json`).

Reviewer: ```suggestion
1. **Placeholder files**. Merchants can change the storefront labels by changing the values in the placeholder JSON files, which are organized by drop-in components—`cart.json`, `checkout.json`, `pdp.json`, and so on.
```

## Step-by-step

> In the boilerplate code, the UI text labels in drop-in components come from the `placeholders` files. Using these files as the source for all storefront UI labels across drop-in components makes it easy for merchants to make label changes without developer involvement.

Reviewer: ```suggestion
In the boilerplate code, the UI text labels in drop-in components come from the placeholder files. By using these files as the source for all storefront UI labels, merchants can easily change labels without involving developers.
```

> There are two key aspects to using `fetchPlaceholders`:

Reviewer: ```suggestion
There are two things to be aware of when using the `fetchPlaceholders()` function:
```

</Task>

<Task>
### Initialize placeholders with path

> During initialization, use `fetchPlaceholders` with the path to your drop-in's placeholders file. This fetches and merges the placeholders into the global object.

Reviewer: ```suggestion
During initialization, you must use the `fetchPlaceholders()` function using an argument to the path to your drop-in's placeholders file. This fetches and merges the placeholders into the global object.
```

```javascript
// Initialize placeholders for this drop-in
const placeholders = await fetchPlaceholders('placeholders/cart.json');

const langDefinitions = {
  default: {
    ...placeholders,
  },
};

// Register Initializers
initializers.mountImmediately(initialize, {
  langDefinitions,
  //...
});
```
</Task>

<Task>
### Fetch placeholders after initialization

> After initialization, you can use `fetchPlaceholders` without a path to retrieve all merged placeholders. The following diagram and code snippet shows how to fetch the placeholders.

Reviewer: ```suggestion
After initialization, you can use the `fetchPlaceholders` function without a path to retrieve all merged placeholders. The following diagram and code snippet shows how to fetch the placeholders.
```
